### PR TITLE
Add identifiers field

### LINKED ISSUE #72: Update schema to reflect v1.1.0
Fixing this issue includes:

- Updating the versions in `schema.yaml`.
- Incorporating schema changes from #92 and #85.

### LINKED ISSUE #85: Add identifier term that isn't a doi
Hello,
I wanted to recommend the usage of CITATION.cff in academic workflows, but I was surprised that there isn't any term for software identification that isn't a DOI.
Even if the usage of a DOI is common and is suggested as a best practice by the SCIWG, there are other ways to identify software, for example:
Wikidata entity identifiers (Q828742 for Scilab at https://www.wikidata.org/wiki/Q828742)
ascl identifiers  (ascl:1908 at http://ascl.net/1905.018)
swMath identifier (swMath-id:834 for Scilab https://swmath.org/software/834)
handle
sha1
and the swh identifier (hashed and resolved by Software Heritage)
Here is the last version captured on SWH:
`swh:1:rev:91fbfb81b72b3d09175fec6a06a5567dac820bd5;origin=https://github.com/scilab/scilab`

The panorama of identifiers is broad and will be discussed on the joint FORCE11 & RDA Software Source Code Identification Working Group [SCID WG](https://github.com/force11/force11-rda-scidwg)
I would like to propose having a term that can contain a list of identifiers, this will enable capturing more metadata and linking it accordingly on different platforms.

What do you think?

### PR DESCRIPTION
**Pull request by sdruskat**
This PR fixes #85 as described in https://github.com/citation-file-format/citation-file-format/issues/85#issuecomment-552432806.

EDIT: Also fixes #72.

### CODE REVIEW

**Comment by sdruskat** [comment]
Test will fail as cffconvert cannot find 1.1.0 schema (yet).

**Comment by jspaaks** [comment]
- [x] verified that the tests work as expected using this approach https://github.com/citation-file-format/citation-file-format/pull/95#issuecomment-552468217
- [ ] checked documentation

File: README.md [hunk 1]
```diff
@@ -840,9 +842,40 @@ file authors to find the optimal name split in any given case.
...
+
+An identifier object represents a persistent identifier. In CFF, identifier objects are realized as
+collections with two defined keys, both mandatory.
+
+| Identifier key      | Identifier data type     | optional       |
+| -               | -                    | :------------: |
+| `type`  | String ([*Identifier type string*](#identifier-type-strings))              |                |
+| `value`   | String               |                |
+
+Table: Complete list of keys for identifier objects.
```
Comment (jspaaks) [inline README.md]: We can probably remove this, given that we have https://github.com/citation-file-format/citation-file-format/pull/96/files#diff-04c6e90faac2675aa89e2176d2eec7d8R955-R956 already.

Comment (jspaaks) [reply README.md]: Wait I just realized it is the table caption.

**Comment by sdruskat** [comment]
Currently unable to run a quick pykwalify test as you had for #95 (https://github.com/citation-file-format/citation-file-format/pull/95#issuecomment-552468217) (on the road), but if it works, feel free to merge @jspaaks.

**Comment by sdruskat** [comment]
Guess merging this also fixes #72.

File: schema.yaml [hunk 2]
```diff
@@ -616,3 +622,16 @@ schema;reference:
...
       type: int
+
+schema;identifier:
+  type: map
+  mapping:
+
+    type:
+      required: True
+      type: str
+      enum: ['software-heritage', 'other']
```
Comment (moranegg) [inline schema.yaml]: I can suggest a longer enum of identifier types with:
```
['ASCL-ID', 'HAL-ID', 'SwMath-ID', 'Wikidata entity', 'FSF directory ID']
```
or package identifier would depend on the package manager:
```
['NPM', 'PyPI', 'Maven']
```

**Comment by moranegg** [comment]
I'm a bit in delay with this merge, but thanks for including me in the discussion..
Just commented the enum of types, because Software Heritage might not be the best example of known identifiers :-)

**Comment by sdruskat** [comment]
Thanks Morane! We'll ask people at the Registries workshop today for suggestions on how to include their own identifiers as well. And will look at how to include those from your list that are missing after that.

**Pull request merged**